### PR TITLE
fix(api): use separate model for plugin scanner state

### PR DESCRIPTION
**Pull request by ramizpolic**
## Description

Use independent model for plugin scanner state.

## Type of Change

[x] Bug Fix  
[ ] New Feature  
[ ] Breaking Change  
[x] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: plugins/store/kics/main.go [hunk 1]
```diff
@@ -111,32 +111,32 @@ func (s *Scanner) Start(config types.Config) {
...
 		if err != nil {
 			logger.Error("Failed to format KICS output", slog.Any("error", err))
-			s.SetStatus(types.NewScannerStatus(types.Failed, types.Ptr(fmt.Errorf("failed to format KICS output: %w", err).Error())))
+			s.SetStatus(types.NewScannerStatus(types.StateFailed, types.Ptr(fmt.Errorf("failed to format KICS output: %w", err).Error())))
 			return
 		}
 
 		logger.Info("Scanner finished running.")
-		s.SetStatus(types.NewScannerStatus(types.Done, types.Ptr("Scanner finished running.")))
+		s.SetStatus(types.NewScannerStatus(types.StateFailed, types.Ptr("Scanner finished running.")))
```
Comment (csatib02) [inline plugins/store/kics/main.go:139]:
```suggestion
		s.SetStatus(types.NewScannerStatus(types.StateDone, types.Ptr("Scanner finished running.")))
```

**Review by csatib02 — APPROVED** [review]
LGTM!

**Review by akijakya — APPROVED** [review]
LGTM

File: plugins/store/kics/main.go (hunk 1 above)
Comment (ramizpolic) [reply plugins/store/kics/main.go]: good catch!

**Pull request merged**